### PR TITLE
Test Python version, and warn/error if not supported

### LINKED ISSUE #117: Is Python 3.2 supported?  pyldap uses missing symbol _PyUnicode_AsUTF8
Hello!

It seems like pyldap does not support Python 3.2 (and presumably lower Python 3), because it is using the symbol `_PyUnicode_AsUTF8`.  At least from my perspective, I don't have a need for Python 3.2 or lower.  But still, the README and `setup.py` don't seem to mention that Python 3 needs 3.3 or later, so I wanted to report this!

Here's how I set up a Python 3.2 environment to verify this.  I'm running on macOS 10.12:

```
# If using MacPorts, bring in Python 3.2 and OpenLDAP.
# If not using MacPorts, then this assumes Python 3.2 is `python3.2`,
# and that the OpenLDAP headers are installed in system paths.
sudo port install openldap python32
TMPDIR=$(mktemp -d)
cd $TMPDIR
curl https://pypi.python.org/packages/47/15/7f9aabb1ce09d3dc5ecdff2f5a97e0a47f724fe8e398eff6a88626d0164c/setuptools-5.5.1.tar.gz#md5=86bbbfa732c234535316a7d74a49c6ad | tar -xzf -
curl https://pypi.python.org/packages/e4/3a/43fbf0c2ed567efd409d19968da3f373f83b99831090e3660be937f645a6/pyldap-2.4.37.tar.gz#md5=55b3b39d9c1c7837d80203a04d52c6a1 | tar -xzf -
cd setuptools-5.5.1
python3.2 setup.py build ; python3.2 setup.py install --user
cd ../pyldap-2.4.37

# The next two lines are only needed when MacPorts installed OpenLDAP.
# Otherwise, it tries to use OS LDAP headers, which doesn't work.
sed -i -e 's|library_dirs =|library_dirs = /opt/local/lib|' setup.cfg
sed -i -e 's|include_dirs =|include_dirs = /opt/local/include|' setup.cfg

python3.2 setup.py build ; python3.2 setup.py install --user
cd ..
rm -rf setuptools-5.5.1 pyldap-2.4.37
cd ~
rmdir $TMPDIR
```

The build and install goes fine; the error is triggered when loading `ldap`:

```
Karls-Mac-Mini-9346:~ akkornel$ python3.2
Python 3.2.6 (default, Oct 11 2016, 14:54:14) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.38)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import ldap
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/akkornel/Library/Python/3.2/lib/python/site-packages/pyldap-2.4.37-py3.2-macosx-10.12-x86_64.egg/ldap/__init__.py", line 22, in <module>
    import _ldap
ImportError: dlopen(/Users/akkornel/Library/Python/3.2/lib/python/site-packages/pyldap-2.4.37-py3.2-macosx-10.12-x86_64.egg/_ldap.so, 2): Symbol not found: _PyUnicode_AsUTF8
  Referenced from: /Users/akkornel/Library/Python/3.2/lib/python/site-packages/pyldap-2.4.37-py3.2-macosx-10.12-x86_64.egg/_ldap.so
  Expected in: flat namespace
 in /Users/akkornel/Library/Python/3.2/lib/python/site-packages/pyldap-2.4.37-py3.2-macosx-10.12-x86_64.egg/_ldap.so
>>> ^D
```

It seems that `PyUnicode_AsUTF8` is present starting in Python 3.3; it is not listed in the equivalent page for Python 3.2.

https://docs.python.org/3.3/c-api/unicode.html#PyUnicode_AsUTF8
https://docs.python.org/3.2/c-api/unicode.html

In terms of current active releases, I'm guessing that only Debian wheezy would really be affected (it ships with Python 3.2).

### PR DESCRIPTION
**Pull request by akkornel**
Two commits.  The first takes @hroncok's suggestion and adds specific classifiers, showing the Pythons we know work.

The second commit tests the Python version, and errors out if running Python 3.0-3.2 inclusive.  Pythons 2.6 and below get a warning.  I chose `RuntimeError` because I saw setuptools use it, and I chose `Warning` because I couldn't find anything more specific.

Also, we set the `python_requires` field, excluding versions <2.7 and 3.0.x-3.2.x.  The <2.7 exclusion is something that could be removed, because of people running 2.6.

Closes #117.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -15,6 +15,11 @@
 except ImportError:
   from distutils.core import setup, Extension
 
+if sys.version_info[0] == 2 and sys.version_info[1] < 7:
+    raise Warning('This software is not tested on your Python.  Be careful!')
```
Comment (hroncok) [inline setup.py:19]: I'd say: "on your Python version"

**Comment by rbarrois** [comment]
Thanks!

I'm unsure though, do we need both the warning/error, *and* the python_requires flag?
I'm pretty sure `pip install` handles the `python_requires` flag properly, and refuses to install when the active Python is incompatible.

**Comment by hroncok** [comment]
I believe the warning/error is for older setuptools and python_requires is the right way to do it. By keeping only the second, it won't work on older setuptools (and, chances are, if the users run 3.2, they have old setuptools as well). If wee only keep the warning/error, it would work, but we would not benefit from python_requires feature, such as, maybe, some nice metadata display on new PyPI?

**Comment by akkornel** [comment]
In my testing, I was working with Python 3.2.3 and setuptools 0.6.24, because those are what Debian wheezy uses.

setuptools 0.6.24 does not recognize the python_requires setting, throws a warning, and then continues.  The build and install took place without errors, and I only got an error at runtime.

So, to deal with really old setuptools, you need the first check.  And I think people using really old Python will also be using really old setuptools.  And they might not be using pip client at all.

File: setup.py [hunk 2]
```diff
@@ -15,6 +15,11 @@
...
+if sys.version_info[0] >= 3 and sys.version_info[1] < 3:
+    raise RuntimeError('The C API from Python 3.3+ is required.')
```
Comment (encukou) [inline setup.py:21]: This would fail on Python 4.0; it should be `if sys.version_info[0] >= 3 and sys.version_info < (3, 3):`

**Comment by encukou** [comment]
Rebased, fixed up a bit, and pushed as 2ae692c .

Thank you for the contribution!

**Pull request closed without merging**